### PR TITLE
Register email addresses with SendGrid

### LINKED ISSUE #804: SendGrid Update Service
Publishers will maintain a list at the Brave SendGrid account. When publisher records are created or updated this service will ensure the data kept at SendGrid will be updated. This needs to support new records, and changes to email addresses and contact information, such as names.

This service will need to communicate with the SendGrid API. SendGrid has the sendgrid-ruby gem (https://github.com/sendgrid/sendgrid-ruby) which provides a ruby interface to the the API and looks like a good place to start.

### PR DESCRIPTION
**Pull request by lgebhardt**
Uploads publishers as contacts to SendGrid as they are created and edited. Also adds them to a specified list. When emails are changed the old email is removed from the list and the new one added.

Includes a rake task to upload all email verified publishers to SendGrid. This rake task is safe to run more than once.

Fixes #804, #805

Submitter Checklist:

- [X] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [X] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [X] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Tagged reviewers.

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

**Comment by lgebhardt** [comment]
To install this on Production and Staging we should have two separate SendGrid accounts. 

The setup steps needed are:
- Create SendGrid account
- Generate an API key under [Settings / API Keys](https://app.sendgrid.com/settings/api_keys) with full access. Give it a name and record the secret key
- At Heroku add a config var for SENDGRID_API_KEY (recorded above)
- run the rake task `sendgrid:init` and record the list id it creates or returns
- At Heroku add a config var for SENDGRID_PUBLISHERS_LIST_ID (list id recorded above)

File: lib/send_grid/api_helper.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        if json_result['recipient_count'] == 0
+          raise NotFoundError.new
+        end
+
+        json_result['recipients'][0]
+      end
+
+      def get_contact_by_email(email:)
+        result = sg.client.contactdb.recipients._(id_from_email(email)).get
+        check_result(result: result, sucess: '200')
```
Comment (yachtcaptain23) [inline lib/send_grid/api_helper.rb]: `success`

File: lib/send_grid/api_helper.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          raise NotFoundError.new(result.body)
+        when '429'
+          raise SendGrid::RateLimitExceeded.new
+        else
+          raise "Unknown Error"
+        end
+      end
+
+      def sg
+        SendGrid::API.new(api_key: Rails.application.secrets[:sendgrid_api_key])
```
Comment (yachtcaptain23) [inline lib/send_grid/api_helper.rb]: Would you cache this value?
`@sg ||= SendGrid::API.new(api_key: Rails.application.secrets[:sendgrid_api_key])`

File: lib/send_grid/api_helper.rb (hunk 1 above)
Comment (lgebhardt) [reply lib/send_grid/api_helper.rb]: Fixed

**Pull request merged**